addons/website_sale/models/website.py

```diff
@@ -516,6 +519,33 @@ def _get_product_page_grid_image_classes(self):
         return spacing_map.get(self.product_page_image_spacing) + ' ' +\
                 columns_map.get(self.product_page_grid_columns)
 
+    @api.model
+    def _send_abandoned_cart_email(self):
+        for website in self.search([]):
+            if not website.send_abandoned_cart_email:
+                continue
+            SaleOrder = self.env['sale.order']
+            abandoned_carts = SaleOrder.search([
+                ('is_abandoned_cart', '=', True),
+                ('cart_recovery_email_sent', '=', False)
+            ])
+            abandoned_datetime = datetime.utcnow() - relativedelta(hours=website.cart_abandoned_delay)
+
+            sales = SaleOrder.search([
+                ('state', '=', 'sale'),
+                ('partner_id', 'in', abandoned_carts.partner_id.ids),
+                ('create_date', '>=', abandoned_datetime)
+            ])
+            abandoned_carts.filtered(lambda so: so._can_send_abandoned_cart_mail(sales))
```

Comment:
Something I noticed, since we are filtering the abandoned carts on those condition without marking them as `sent`, the amount of sale order we will have to check will only grow with time.
```suggestion
            abandoned_carts.filtered(lambda so: so._can_send_abandoned_cart_mail(sales))
            (sales - abandoned_carts).cart_recovery_email_sent = True
```
That way abandoned_carts that do not meet are requirements will not be checked again.
Maybe this needs further checks but I am not sure at the moment.

Also maybe we need to reset the cart_recovery_email_sent flag if there is a change to the cart.